### ISSUE TITLE
Vangen exits with status 0 when there is a syntax error

### ISSUE THREAD

**Issue by hillna** [leighmcculloch/vangen#14]
When there is a syntax error in the vangen.json file, vangen will output the error and then exit with status 0.

```bash
> go run 4d63.com/vangen
 invalid character ']' looking for beginning of value
> echo $?
 0
```

We have a pipeline that utilizes vangen which will still pass as successful when vangen fails.

Executable should exit with status 1 when there is a syntax error in the configuration.

**Comment by leighmcculloch** [comment]
Good catch. This should definitely be causing a non-zero exit status. It looks like there are a few error conditions where this is happening in the application.

**Comment by chrsm** [comment]
Hi @leighmcculloch - I added pr#15 to address this - as well as other error cases where main would return after printing an error. Let me know if this is inline with your style/etc; I am happy to make any changes you'd like.

**Comment by leighmcculloch** [comment]
The fix created by @chrsm is released in [v1.3.0](https://github.com/leighmcculloch/vangen/milestone/2).
